examples/userapi/04_boundary_conditions.ipynb

```diff
@@ -0,0 +1,530 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Boundary conditions tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial aims to demonstrate how users can implement various boundary conditions in Devito, building on concepts introduced in previous tutorials. Over the course of this notebook we will go over the implementation of both free surface boundary conditions and perfectly-matched layers (PMLs) in the context of the first-order acoustic wave equation. This tutorial is based on a simplified version of the method outlined in Liu and Tao's 1997 paper (https://doi.org/10.1121/1.419657).\n",
```

Comment:
simplified version of PML isn't really PML. I know I'm beeing picky but saying Devito can do PML and having reflections as strong as the result at the end does not send a great message. People expect near perfect absorption with PML not damping layer quality.

Reply:
Could you send me an example of a non-simplified PML? As far as I can tell, my implementation is equivalent to any that I've seen in the literature (excluding CPMLs), and also near-identical to one that an industry guy showed me at OGHPC. The main difference is I'm pretty sure the tapers used are somewhat sub-optimal (because few papers actually explicitly state what tapers they used), which will affect the reflections.

Reply:
The paper below is simplified PML, but here is a simplified version of simplified PML, like most of the f1/f2/f3/f4/f5 from the equations are missing. So yes, most are simplifications in some ways of CPML, and I don't say I have an issue with the implementation or the tutorial, just would have worded it in a more cautious way .

Reply:
Should they not just fall out if you print the update equation? f1/2/3/etc are just assorted bits of finite difference stencil right?

Reply:
Inside the domain yes, but in the layer it depends on the omega's so it doesn't.

Reply:
Iirc, the omega_etas inside f1/2/3/etc are just badly named damping parameters and don't have anything to do with the frequency

Reply:
Yes they are, but they are still part of the stencil inside the layer, while only f1 is there currently